model/riscv_insts_zcmp.sail

```diff
@@ -0,0 +1,643 @@
+  (0x9, 0b10) <-> "{ra, s0-s4}, -64",
+  (0x9, 0b11) <-> "{ra, s0-s4}, -80",
+  (0xa, 0b00) <-> "{ra, s0-s5}, -32",
+  (0xa, 0b01) <-> "{ra, s0-s5}, -48",
+  (0xa, 0b10) <-> "{ra, s0-s5}, -64",
+  (0xa, 0b11) <-> "{ra, s0-s5}, -80",
+  (0xb, 0b00) <-> "{ra, s0-s6}, -32",
+  (0xb, 0b01) <-> "{ra, s0-s6}, -48",
+  (0xb, 0b10) <-> "{ra, s0-s6}, -64",
+  (0xb, 0b11) <-> "{ra, s0-s6}, -80",
+  (0xc, 0b00) <-> "{ra, s0-s7}, -48",
+  (0xc, 0b01) <-> "{ra, s0-s7}, -64",
+  (0xc, 0b10) <-> "{ra, s0-s7}, -80",
+  (0xc, 0b11) <-> "{ra, s0-s7}, -96",
+  (0xd, 0b00) <-> "{ra, s0-s8}, -48",
+  (0xd, 0b01) <-> "{ra, s0-s8}, -64",
+  (0xd, 0b10) <-> "{ra, s0-s8}, -80",
+  (0xd, 0b11) <-> "{ra, s0-s8}, -96",
+  (0xe, 0b00) <-> "{ra, s0-s9}, -48",
+  (0xe, 0b01) <-> "{ra, s0-s9}, -64",
+  (0xe, 0b10) <-> "{ra, s0-s9}, -80",
+  (0xe, 0b11) <-> "{ra, s0-s9}, -96",
+  (0xf, 0b00) <-> "{ra, s0-11}, -64",
+  (0xf, 0b01) <-> "{ra, s0-11}, -80",
+  (0xf, 0b10) <-> "{ra, s0-11}, -96",
+  (0xf, 0b11) <-> "{ra, s0-11}, -112",
+}
+
+mapping CM_PUSH_MAPPING_64 : (bits(4), bits(2)) <-> string = {
+  (0x4, 0b00) <-> "{ra}, -16",
+  (0x4, 0b01) <-> "{ra}, -32",
+  (0x4, 0b10) <-> "{ra}, -48",
+  (0x4, 0b11) <-> "{ra}, -64",
+  (0x5, 0b00) <-> "{ra, s0}, -16",
+  (0x5, 0b01) <-> "{ra, s0}, -32",
+  (0x5, 0b10) <-> "{ra, s0}, -48",
+  (0x5, 0b11) <-> "{ra, s0}, -64",
+  (0x6, 0b00) <-> "{ra, s0-s1}, -32",
+  (0x6, 0b01) <-> "{ra, s0-s1}, -48",
+  (0x6, 0b10) <-> "{ra, s0-s1}, -64",
+  (0x6, 0b11) <-> "{ra, s0-s1}, -80",
+  (0x7, 0b00) <-> "{ra, s0-s2}, -32",
+  (0x7, 0b01) <-> "{ra, s0-s2}, -48",
+  (0x7, 0b10) <-> "{ra, s0-s2}, -64",
+  (0x7, 0b11) <-> "{ra, s0-s2}, -80",
+  (0x8, 0b00) <-> "{ra, s0-s3}, -48",
+  (0x8, 0b01) <-> "{ra, s0-s3}, -64",
+  (0x8, 0b10) <-> "{ra, s0-s3}, -80",
+  (0x8, 0b11) <-> "{ra, s0-s3}, -96",
+  (0x9, 0b00) <-> "{ra, s0-s4}, -48",
+  (0x9, 0b01) <-> "{ra, s0-s4}, -64",
+  (0x9, 0b10) <-> "{ra, s0-s4}, -80",
+  (0x9, 0b11) <-> "{ra, s0-s4}, -96",
+  (0xa, 0b00) <-> "{ra, s0-s5}, -64",
+  (0xa, 0b01) <-> "{ra, s0-s5}, -80",
+  (0xa, 0b10) <-> "{ra, s0-s5}, -96",
+  (0xa, 0b11) <-> "{ra, s0-s5}, -112",
+  (0xb, 0b00) <-> "{ra, s0-s6}, -64",
+  (0xb, 0b01) <-> "{ra, s0-s6}, -80",
+  (0xb, 0b10) <-> "{ra, s0-s6}, -96",
+  (0xb, 0b11) <-> "{ra, s0-s6}, -112",
+  (0xc, 0b00) <-> "{ra, s0-s7}, -80",
+  (0xc, 0b01) <-> "{ra, s0-s7}, -96",
+  (0xc, 0b10) <-> "{ra, s0-s7}, -112",
+  (0xc, 0b11) <-> "{ra, s0-s7}, -128",
+  (0xd, 0b00) <-> "{ra, s0-s8}, -80",
+  (0xd, 0b01) <-> "{ra, s0-s8}, -96",
+  (0xd, 0b10) <-> "{ra, s0-s8}, -112",
+  (0xd, 0b11) <-> "{ra, s0-s8}, -120",
+  (0xe, 0b00) <-> "{ra, s0-s9}, -96",
+  (0xe, 0b01) <-> "{ra, s0-s9}, -112",
+  (0xe, 0b10) <-> "{ra, s0-s9}, -128",
+  (0xe, 0b11) <-> "{ra, s0-s9}, -144",
+  (0xf, 0b00) <-> "{ra, s0-s11}, -112",
+  (0xf, 0b01) <-> "{ra, s0-s11}, -128",
+  (0xf, 0b10) <-> "{ra, s0-s11}, -144",
+  (0xf, 0b11) <-> "{ra, s0-s11}, -160",
+}
+
+mapping CM_POP_MAPPING_32 : (bits(4), bits(2)) <-> string = {
```

Comment:
Push and pop would ideally be the same mapping, just with the adjustment negated, not duplicated